utils/nlp.py

```diff
@@ -1,30 +1,53 @@
-from typing import List
-import spacy
-import nltk
-from nltk.corpus import stopwords
-
-
-def text_preprocess(text: str, lang: str = "english") -> List[str]:
-    """
-    Does some preprocessing to better analyze a text.
-    Lowercase, remove punctuation and stopwords, tokenization and lemmatization
-
-    Returns:
-    List of processed tokens
-    """
-    if lang == "english":
-        nlp = spacy.load("en_core_web_md")
-    elif lang == "french":
-        nlp = spacy.load("fr_core_news_md")
-    else:
-        raise NotImplementedError()
-    
-    
-    doc = nlp(text)
-    pos_lemmas = [[token.pos_, token.lemma_] for token in doc]
-    
-    # Remove stop words and punctuation
-    stop_words = set(stopwords.words(lang))
-    word_tokens = [w[1] for w in pos_lemmas if w[1] not in stop_words and w[0] != 'PUNCT']
-    
-    return word_tokens
+from typing import Optional, Tuple
+from langdetect import DetectorFactory, detect, LangDetectException
+from transformers import pipeline, TFRobertaForSequenceClassification, TFCamembertForSequenceClassification, AutoTokenizer
+
+from models.comment import SentimentEnum
+
+DetectorFactory.seed = 0
+
+def detect_language(text: str) -> str:
+    try:
+        return detect(text)
+    except LangDetectException:
+        return 'unknown'
+
+class SentimentAnalysis:
+    MODELS_FOLDER = "./data/sentiment_models"
+
+    def __init__(self, config):
+        self.analysis_enabled = config.get("use_sentiment_analysis")
+        if self.analysis_enabled:
+            model_en = TFRobertaForSequenceClassification.from_pretrained(self.MODELS_FOLDER + "/english")
```

Comment:
You can use the os.path.join function to join paths. This ensures platform-independent path joining and avoids potential issues with path separators.

Reply:
Done